src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -393,19 +393,53 @@ impl ScalarFunc {
         Ok(Some(Cow::Owned(MyDuration::zero())))
     }
 
+    #[inline]
+    pub fn add_date_and_duration<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let arg0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
```

Comment:
Seems there is a bug in `TiDB`(https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L4298) where we should eval a date for the first argument. @zz-jason

Reply:
@@XuHuaiyu

Reply:
yeah, you are right, since the first parameter is `TypeDate`, whose eval type is `ETDate`, `EvalTime()` should be called.

Reply:
It seems that the SubDateAndXXX functions are also wrong.

Reply:
@koushiro Would you like to file a PR to tidb repo to fix these issues?